FirebaseDatabase/Sources/Core/FSyncPoint.m

```diff
@@ -199,6 +205,15 @@ - (NSArray *)addEventRegistration:(id<FEventRegistration>)eventRegistration
     return [view initialEvents:eventRegistration];
 }
 
+- (BOOL)unregisterQuery:(FQuerySpec *)query {
```

Comment:
This method just removes the query from the view map if no other query added an event registration to it.

There is quite a bit of logic that isn't brought over from `removeEventRegistration`. In particular, we are not:

- We are not removing all empty views from a syncpoint if the getData query is default.
- We are not generating any `FCancelEvent`s for the query since there is no event registration associated with it.

I'm least sure about this function.